Allow more language codes

rymai opened the pull request:
Related to #12.

Per Transifex documentation (https://www.transifex.com/languages/), language
codes in the form `en` (2 chars) / `ach` (3 chars) and `en_US` ("_" separator)
are allowed, so let's allow them here too!

I hope that's an acceptable change?

Cheers!

rymai committed:
Allow more language codes

Per Transifex documentation (https://www.transifex.com/languages/), language
codes in the form `en` (2 chars) / `ach` (3 chars) and `en_US` ("_" separator)
are allowed, so let's allow them here too!

tsido commented on tasks/transifex-resjson-tasks.js:
```diff
@@ -720,7 +720,7 @@ module.exports = function (grunt) {
     */
     function mapToTxLangCode(str) {
-        var matcher = str.match(/([a-z]{2})-([A-Z]{2}|latn)$/i);
+        var matcher = str.match(/([a-z]{2,3})(?:[-_]([a-z]{2}|latn))?$/i);
```
Why include the underscore (`_`) as accepted separator in the regexp? Otherwise this looks like a easy way to improve the mappings quickly.

rymai replied:
I thought it would not hurt to accept the right format directly (so that people can name their folder after Transifex's nomenclature if they prefer (e.g. `en_US`)?

tsido replied:
Accepting the underscore will break things. When doing the mapping back from Transifex language codes to the file system,  Transifex's code `en_US` will be converted to `en-US` by `mapFromTxLangCode()` even though it originally was `en_US`. Accepting it would require additional changes for handling the mappings, so better leave it out here.

Otherwise happy to merge this in, after that small fix!

rymai replied:
Thanks for the explanation @tsido! I've removed `_` from the accepted separators then!

rymai committed:
Remove '_' from the accepted separator for language codes

Per @tsido's comment:
"Accepting the underscore will break things. When doing the mapping back from Transifex language codes to the file system, Transifex's code en_US will be converted to en-US by mapFromTxLangCode() even though it originally was en_US. Accepting it would require additional changes for handling the mappings, so better leave it out here."
